Update pipeline-build-template.yaml

emuthiah opened the pull request:
* Updated git secret and webhooks
* Updated sonarqube settings and added example command

emuthiah committed:
Update pipeline-build-template.yaml

* Updated git secret and webhooks
* Updated sonarqube settings and added example command

emuthiah committed:
Update pipeline-build-template.yaml

* corrected example sonar-scanner command

ChrisHoban reviewed (commented):
I don't think we should switch to only supporting Jenkins Credentials for Git Checkout, we should be storing these credentials in OpenShift as that is what will be backed by Vault

emuthiah committed:
Update pipeline-build-template.yaml

* Added openshift secret to store git private ssh key

ChrisHoban commented:
Something to note, I have minorly overhauled the file you are working with last week :)

https://github.com/ChrisHoban/ssg-openshift-meta-templates/commit/92871a710d58f2d0c4fb9088ad1a67f1cd1eb684

emuthiah commented:
Looks like we both have been working on it simultaneously :-)  
Totally agree with you, the secrets should be in openshift. I've updated the template to store the credentials in openshift. 
To sync openshift secrets with jenkins credentials: https://github.com/openshift/jenkins-client-plugin/blob/master/README.md#setting-up-credentials

emuthiah commented:
The secret is just a place holder with the value 'set.ssh-private-key'

ChrisHoban commented:
Ahh, gotcha. I guess last question. I am kind of operating as the "Product Owner" to this, and am totally open to improvements and feature additions etc, but I am a bit lost on this one. What is the ultimate goal you are working towards? Is it in the Features/Requests? Could it be added? I guess there are 2 things I am reviewing on these. 1. Are the changes done well or is there some feedback to square them up and 2. Is whats being changed/enhanced aligned with the product vision in some way. I guess I want to know what your goals are regarding the second point, and trying to make sure this is a "For everyone" kind of enhancement/change and not aimed at something you are working on in one project? :)

emuthiah committed:
update pipeline-build-template.yaml

* bug fix: corrected the git credentialsid:
  https://github.com/openshift/jenkins-client-plugin/blob/master/README.md#setting-up-credentials
  This token will be accessible with the credential ID of "${OPENSHIFT_NAMESPACE}-my-prilvileged-token-id"

emuthiah commented:
For sure! Sorry I did not give details for the PR. There are two changes:
1. Git
This applies to everyone, It is a fix/enhancement of the template. When we use private git repos, we have to supply git credentials to Jenkins git plugin, AFAIK this can only be done using Jenksins credentialsId (by adding the ssh key to jenkins credential store and using the jenksins credentialsId with the git plugin). Previously, GIT_CREDENTIALS_ID was set using gitSecret (random gen.).
2. Sonarqube 
This also applies to everyone. I've commented the example command so it is not executed if not needed. To run sonar-scanner we only need the project key and token, sonarJenkinsCredentialId is not needed.
